### ISSUE TITLE
Enable searching for packages with topics with `#<topic-name>`

### ISSUE THREAD

**Issue by parlough** [dart-lang/pub-dev#7080]
Since topics are presented as [hashtags](https://en.wikipedia.org/wiki/Hashtag), I believe users expect to be able to search for them in the same way since other platforms with hashtags do as well.

It's otherwise not obvious how to search for a topic (`topic:<name>`), as it's not an option in the left form. Users will only learn by reading the documentation or by clicking on a topic. It should be instantly intuitive and relying on the well-known search behavior of hashtags from other platforms should provide that :) 

I would expect specifying more than one would topic would require them all to be specified by the listed packages. That way users can dive down into a topic subcategory. For example `#server #html`.

This should be verified to not break quoted `"#<topic name>"` literal text search.

**Comment by parlough** [comment]
\cc @szakarias For your thoughts :)

**Comment by szakarias** [comment]
Great idea!

**Comment by parlough** [comment]
Resolved by https://github.com/dart-lang/pub-dev/pull/8227
